File: 35-SearchInsertPosition/35-SearchInsertPosition.java
// Last updated: 9/16/2025, 12:53:29 AM
class Solution {
    public int searchInsert(int[] nums, int target) 
    {
        int len = nums.length;
        int l = 0, r = len-1;
        int mid;
        while(l<=r)
        {
            mid = (l+r)/2;
            if(nums[mid] == target)
                return mid; 
            else if (nums[mid]> target)
            r = mid -1;
            else if (nums[mid]<target)
            l = mid+1;
        }    
        return l;
    }
}

File: 35-SearchInsertPosition/35-SearchInsertPosition.cpp
// Last updated: 2/13/2026, 12:58:25 PM
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) 
    {
        int l = 0, h = nums.size()-1;
        int a = nums.size();
        while(l<=h)
        {
            int m = (l+h)/2;
            if(nums[m] >= target) 
            {
                a = m;
                h = m-1;
            }
            else l = m+1;
        }
        return a;
    }
};